.github/setup_python/action.yml

```diff
@@ -1,27 +1,55 @@
 name: "Setup Python"
 description: "Sets up Python for use in actions with caching and copy localsettings"
 
+inputs:
+  requirements-file:
+    description: "if disabled this will only install runtime dependencies"
+    required: false
+    default: requirements-dev.txt
+
 runs:
   using: "composite"
   steps:
     - name: Setup python
       uses: actions/setup-python@v4
       with:
-        python-version: '3.10'
+        python-version: "3.10"
 
     - name: Restore venv cache
-      uses: syphar/restore-virtualenv@v1
       id: cache-virtualenv
+      uses: actions/cache@v3
+      with:
+        path: .venv
+        key: venv-${{ runner.os }}-${{ hashFiles( inputs.requirements-file ) }}-${{ inputs.requirements-file }}
```

Comment:
I think this doesn't suffice: In `requirements-dev.txt` we have `-r requirements.txt`, so the hash of `requirements-dev.txt` doesn't change if we change the contents of `requirements.txt`. Maybe the
```
with:
  requirements-file: requirements.txt
```
interface is too broad afterall and a boolean argument `install-dev-dependencies` is better to clearly state that we are always talking about `requirements.txt` and `requirements-dev.txt` and nothing else. The easiest cache key would then be to always hash both files

Reply:
nicely spotted. Will do so...

Reply:
-dev.txt only: 0c4dd49f29fc31a89010887dc58c0ec0c4b0253d7d12bc8cce03363be6176822
both: 3884987a67ff81e209abc66c7d6d8b7c8e79818064cabd67ec7fa78cb9edb069